backend/dataall/modules/s3_datasets_shares/aws/lakeformation_client.py

```diff
@@ -160,16 +164,29 @@ def _grant_permissions_to_resource(
                     if permissions_with_grant_options:
                         grant_dict['PermissionsWithGrantOption'] = permissions_with_grant_options
 
-                    response = self._client.grant_permissions(**grant_dict)
-
-                    log.info(
-                        f'Successfully granted principal {principal} '
-                        f'permissions {permissions} '
-                        f'and permissions with grant options {permissions_with_grant_options} '
-                        f'to {str(resource)}  '
-                        f'response: {response}'
-                    )
-                    time.sleep(2)
+                    retries = 0
+                    last_error = None
+                    while retries < NUMBER_OF_RETRIES:
+                        try:
+                            response = self._client.grant_permissions(**grant_dict)
+                            last_error = None
+                            log.info(
+                                f'Successfully granted principal {principal} '
+                                f'permissions {permissions} '
+                                f'and permissions with grant options {permissions_with_grant_options} '
+                                f'to {str(resource)}  '
+                                f'response: {response}'
+                            )
```

Comment:
Why do we need this `time.sleep(2)` in line 180? If it succeeds then it could continue without waiting no?